chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+Base metrics are exposed under the path `/metrics?scope=base`.
+
+- *Application Metrics* are specific to an application, they are defined by the developers of the MicroProfile applications themselves. These are custom metrics that are specific to the business logic or operational aspects of the application. Developers use annotations or programmatic APIs to create and register these metrics, tailoring them to monitor the performance and behavior of their application's unique functionalities. Application metrics enable developers to gain insights into the runtime characteristics of their application, such as the number of transactions processed, response times for specific endpoints, or the rate of specific business events. 
+Application metrics are exposed under the path `/metrics?scope=application`.
+
+- *Vendor Metrics* are specific to a particular vendor or technology. These metrics provide insights into the performance and behavior of the runtime's internal components and extensions. Since different MicroProfile implementations may offer additional features or optimize certain areas differently, vendor metrics can vary widely between runtimes. They allow runtime vendors to expose unique metrics that are relevant to their implementation, offering users the ability to monitor vendor-specific aspects of their applications.
+Application metrics are exposed under the path `/metrics?scope=vendor`.
+
+Besides the standard metrics above, MicroProfile Metrics also supports custom metrics. You can use custom metrics to track application-specific information not covered by the standard metrics.
+
+=== Metric Registry
+
+The *MetricRegistry* component acts as a container for storing and managing metrics within an application. It provides a structured way to collect, organize, and access various types of metrics (e.g., counters, gauges, histograms, timers, and metered metrics) for monitoring the behavior and performance of applications. It offers a centralized repository where metrics can be registered, updated, and retrieved. This allows applications to consistently monitor critical operational and performance statistics.
+
+==== Types of Metric Registries
+
+MicroProfile Metrics defines several types of registries, categorized by their scope:
+
+- Application Scope (`MetricRegistry.Type.APPLICATION`): Contains custom metrics that are specific to the application. These are typically the metrics that developers explicitly create and register to monitor application-specific behaviors.
+
+- Base Scope (`MetricRegistry.Type.BASE`): Contains metrics that are fundamental and common across all MicroProfile applications. These metrics provide basic information about the underlying JVM and application server.
+
+- Vendor Scope (`MetricRegistry.Type.VENDOR`): Contains metrics that are specific to the implementation of the MicroProfile platform being used. These metrics offer insights into vendor-specific features and optimizations.
+
```

Comment:
Since custom scopes were mentioned above, it may be worth mentioning that a meter registry is created for these custom scopes.

Reply:
Added